pydm/tests/widgets/test_pushbutton.py

```diff
@@ -0,0 +1,566 @@
+    alarm_sensitive_content : bool
+        True if the widget will change color accordingly to the alarm's severity; False if not
+    """
+    pydm_pushbutton = PyDMPushButton()
+    qtbot.addWidget(pydm_pushbutton)
+
+    if alarm_sensitive_content:
+        pydm_pushbutton.alarmSensitiveContent = alarm_sensitive_content
+
+        signals.new_severity_signal.connect(pydm_pushbutton.alarmSeverityChanged)
+        signals.new_severity_signal.emit(alarm_severity)
+
+        assert pydm_pushbutton._alarm_state == alarm_severity
+        expected_style = dict(test_alarm_style_sheet_map[PyDMWidget.ALARM_CONTENT][alarm_severity])
+        assert pydm_pushbutton._style == expected_style
+    else:
+        assert pydm_pushbutton.alarmSensitiveContent == PyDMWidget.ALARM_NONE
+
+
+# --------------------
+# NEGATIVE TEST CASES
+# --------------------
+
+@pytest.mark.parametrize("current_channel_value, updated_value, expected_log_error", [
+    (np.array([123.123, 456.456]), 10.10, "'10.1' is not a valid pressValue"),
+    (np.array(["abc", "string in an array"]), "New str value", "'New str value' is not a valid pressValue"),
+
+    (10, "New str value", "not a valid"),
+    (10.10, "New str value", "not a valid"),
+])
+def test_update_press_value_incompatible_update_value(qtbot, signals, caplog, current_channel_value, updated_value,
+                                                      expected_log_error):
+    """
+    Test if the widget will log the correct error message if the update value's type is incompatible with the current
+    data type established by the current data associated with the widget.
+
+    Parameters
+    ----------
+    qtbot : fixture
+        Window for widget testing
+    signals : fixture
+        The signals fixture, which provides access signals to be bound to the appropriate slots
+    caplog : fixture
+        The fixture to capture log outputs
+    current_channel_value : int, float, str, ndarray
+        The existing channel value
+    updated_value : int, float, str
+        The new channel to send to the channel by clicking on the widget
+    expected_log_error : str
+        The expected error message in the log
+    """
+    pydm_pushbutton = PyDMPushButton()
+    qtbot.addWidget(pydm_pushbutton)
+
+    # First, set the current channel type
+    signals.new_value_signal[type(current_channel_value)].connect(pydm_pushbutton.channelValueChanged)
+    signals.new_value_signal[type(current_channel_value)].emit(current_channel_value)
+
+    # Verify the new value can be converted/cast as long as the casting can be done
+    signals.new_value_signal[type(updated_value)].connect(pydm_pushbutton.updatePressValue)
+    signals.new_value_signal[type(updated_value)].emit(updated_value)
+
+    # Make sure logging capture the error, and have the correct error message
+    for record in caplog.records:
+        assert record.levelno == logging.ERROR
+    assert expected_log_error in caplog.text
```

Comment:
Wow! Checking logging messages. Definitely don't delete this but this might be a level of testing we don't quite need.

Reply:
The reason for checking the logging messages is to make sure we have the right errors recorded. Do you suggest we check stderr instead?

Reply:
No. You tested it correctly. I was just commenting on the fact that this mighty be overly thorough.